### PR TITLE
APPS/IO_DEMO: Check time limit during connection retries

### PR DESCRIPTION
**Pull request by dmitrygx**
## What

Check time limit during connection retries

## Why ?

Fixes https://github.com/openucx/ucx/pull/5201#issuecomment-653504502

## How ?

Check the run-time limit during connection retries

### CODE REVIEW

**Comment by dmitrygx** [comment]
@amaslenn could you pls try this patch? whether it fixes https://github.com/openucx/ucx/pull/5201#issuecomment-653504502

**Comment by dmitrygx** [comment]
@hoopoepg @brminich could you review pls?

File: test/apps/iodemo/io_demo.cc [hunk 1]
```diff
@@ -678,11 +683,14 @@ class DemoClient : public P2pDemoCommon {
                                  0, opts().operations.size() - 1)];
     }
 
-    inline void check_time_limit(double current_time) {
+    inline bool check_time_limit(double current_time) {
```
Comment (yosefe) [inline test/apps/iodemo/io_demo.cc:686]: maybe no need to return bool: update_retry can check `if (_status != OK)` return true

Comment (dmitrygx) [reply test/apps/iodemo/io_demo.cc:686]: yes, makes sense
done

**Comment by amaslenn** [comment]
> @amaslenn could you pls try this patch? whether it fixes [#5201 (comment)](https://github.com/openucx/ucx/pull/5201#issuecomment-653504502)

Checked, works as expected now, thanks!

**Pull request merged**